### ISSUE TITLE
Handle Error Codes for non-remoting servers

### ISSUE THREAD

**Issue by dansanduleac** [palantir/conjure-java-runtime#618]
`SerializableErrorToExceptionConverter`, which is forced on you by `OkHttpClients` when it adds the `SerializableErrorInterceptor` to the client, [makes it hard to extract the original HTTP status](https://github.com/palantir/http-remoting/blob/3.8.0/error-handling/src/main/java/com/palantir/remoting3/errors/SerializableErrorToExceptionConverter.java) coming from a non-remoting server.

This unnecessarily restricts okhttp-clients' functionality even though we might want to use this when connecting to non-remoting servers too.

Basically same as #56 but in reverse.
